### PR TITLE
feat: Linear map.

### PR DESCRIPTION
**Pull request by daniel-wong-dfinity-org**
This was spun out of sns governance so that it can be used elsewhere (in another PR that I'm working on).

# References

(No ticket.)

[👈 Previous PR][prev] | [Next PR 👉][next]

[prev]: https://github.com/dfinity/ic/pull/2320
[next]: https://github.com/dfinity/ic/pull/2339

### CODE REVIEW

File: rs/nervous_system/linear_map/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+use rust_decimal::Decimal;
+use std::ops::Range;
+
+// DO NOT MERGE - Migrate SNS rewards to this.
+/// A function that maps from one interval to another.
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct LinearMap {
```
Comment (max-dfinity) [inline rs/nervous_system/linear_map/src/lib.rs:7]: Can you add some documentation?  I don't really understand the intended context or application of this structure.  After looking at the tests, I understand what it does, but I'm left with the question "Why do I need this?"

The "Why is this needed?  When is this useful?" becomes harder to answer when it's in its own library, b/c now it's pulled out of the context that justifies it.

Comment (daniel-wong-dfinity-org) [reply rs/nervous_system/linear_map/src/lib.rs:7]: Did you see this triple slash comment? I thought that would be sufficient. I can certainly elaborate (tomorrow).

Comment (daniel-wong-dfinity-org) [reply rs/nervous_system/linear_map/src/lib.rs:21]: Done.

**Review by max-dfinity — APPROVED** [review]
